src/main/java/com/iota/iri/IXI.java

```diff
@@ -87,6 +89,7 @@ private void processWatchEvents() {
                 key = watcher.poll(1000, TimeUnit.MILLISECONDS);
             } catch (InterruptedException e) {
                 log.error("Watcher interrupted: ", e);
+                Thread.currentThread().interrupt();
```

Comment:
👎 
I believe this wasn't the intention originally.
They clearly wanted to not fail. And now you'll probably cause yet another `InterruptedException` by forcing this thread to not die gracefully.

Reply:
Going with this [answer on stackoverflow](https://stackoverflow.com/a/4906814/4604830)  swallowing should only be done if the code implements a interruption policy. I don't know if that's the case here. 
Anyhow, we could leave a comment in the code if swallowing that exception was intentionally.